### PR TITLE
feat: add install script with instructions

### LINKED ISSUE #107: Automation script?
I was looking at the README, and thought about making a shell script to autmate the installation of apaxy. We could have it automatically adjust the .htaccess files depending on the path and all that fancy stuff. I’m not talking about Docker, but an independent script to install it to a given path. Is this something we would make use of?

### PR DESCRIPTION
**Pull request by jordanbancino**
fixes #107 

## proposed changes

- Add an installation script
- Provide instructions on how to use the installation script
- Fix the manual installation instructions to be more general. This will allow us to change things without having to update the readme all the time.

### CODE REVIEW

File: apaxy.config [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Bad : "var/www/html"   <- Does not start with "/", "./", or "../"
+# Good: "./var/www/html" <- Starts with "./", does not have trailing slash
+# Bad : "/var/www/html/" <- Has trailing slash
+
+#
+# GENERAL CONFIGURATION
+#
+
+# The directory to configure Apaxy for use in. This is relative to the web server root, and contains the "theme" folder.
+# For instance: "http://website.com/share" would make this value "/share", where "/share/apaxy/theme" is the theme directory
```
Comment (oupala) [inline apaxy.config]: What should be the value of INSTALL_DIRECTORY if apaxy is used at the root level of the webserver? It should be something like `/`, but it is then not complying the rule above, where it should not have a trailing slash...

Comment (jordanbancino) [reply apaxy.config]: Hmm. I completely overlooked that. Maybe something like `./.`. I'll have to take a look and see.

**Comment by oupala** [comment]
In a first standard test, everything works like a charm. Good point!

I have to test it more deeply, but it looks good!

File: apaxy.config (hunk 1 above)
Comment (oupala) [reply apaxy.config]: Did you had time to have a look at it?

Comment (oupala) [reply apaxy.config]: I tested installing apaxy at the root level of a server, and the configuration is simple just leave the INSTALL_DIRECTORY var empty:

```bash
INSTALL_DIRECTORY=""
```

And it seems to work well!

**Comment by oupala** [comment]
As far as I tested, the install feature works.

But I think this is a problem to link the *configure feature* with the *install feature*.

There is a bunch of usecases where you don't want to install apaxy, you juste want to configure it:

- when you want to configure apaxy, then you have to upload it to the server using ftp, sftp or ftps (when you don't have a console access on your mutualized web space)
- when you already have installed apaxy (ie. using a Dockerfile) and just want to configure it (configure the paths)

Moreover, the files used are named `apaxy.config` and `apaxy-configure.sh`. Hence their job is to configure apaxy, not to install it.

So, in my opinion we should split the two features. Either we have 2 scripts (`apaxy-configure.sh` and `apaxy-install.sh`) and one script can launch the other one (probably the install script should call the configure script). Either we keep the current script, but we add an option (which is *false* by default) to tell if we want to install apaxy or not after configuring it (`--install true` or `--install false`, where false is the default value whenever the parameter is not set on the command line).

What do you think of that @jordanbancino?

It could be good to close this merge request in a near futur so that we can fix the Dockerfile issue (#115) by using this configure script in the Docker image build.

**Comment by jordanbancino** [comment]
Yep, I completely agree with everything you’ve said here. I am so sorry, but I just simply don’t have the time to help out with this project. Maybe over the summer I can take a look again and fix it up. Just not now, or any time soon, I have a lot going on. I do appreciate being mentioned in these, so I can keep up with the project though. I definitely think the changes you’re proposing are important and should be looked at, though.

**Comment by oupala** [comment]
If I understand well, I can try to fix it myself. But if I don't, you might do that in a few months.

Am I right?

**Comment by jordanbancino** [comment]
Yes, that sounds good. I would love to help out, and maybe I can do something every once and a while, there’s just a lot going on for me that I need to figure out first. Summer is the best time for me to get stuff done, though with work and college and stuff, it may be hard.

File: apaxy.config (hunk 1 above)
Comment (jordanbancino) [reply apaxy.config]: Ah cool, glad that works. We will need to make sure that’s documented.

**Comment by oupala** [comment]
Could you please add me to your apaxy repo so that I can add content to your pull request?

Otherwise I can start a new pull request or branch, but all the useful comments from this pull request will miss us.

**Comment by jordanbancino** [comment]
Yep, no problem, I just did. 👍 My schedule should be clearing up just a tad here soon so I may be able to help, but no guarantees yet.

**Comment by oupala** [comment]
I have largely rewrote the install process and I have tester many usecases:

- apaxy installed at the webroot
- apaxy installed somewhere deeper in another directory

Could you please tell me if you have time to test it a bit before I merge the pull request?

**Comment by oupala** [comment]
It seems that you don't time to test this pull request at this moment. I'll merge this pull request as I have other works that depend on this install script.

Feel free to test it later and to open issues if you find any problem.

**Pull request merged**